Add documentation for Jira Helm chart (#13730)
gravitational/teleport: docs/pages/setup/helm-reference/teleport-plugin-jira.mdx

---
title: teleport-plugin-jira Chart Reference
description: Values that can be set using the teleport-plugin-jira Helm chart
---

> The `teleport-plugin-jira` Helm chart is used to configure the Jira Teleport plugin which allows users to receive and manage access requests as tasks in a Jira project.

Reviewer: ```suggestion
The `teleport-plugin-jira` Helm chart is used to configure the Jira Teleport plugin, which allows users to receive and manage Access Requests as tasks in a Jira project.
```

You can [browse the source on GitHub](https://github.com/gravitational/teleport-plugins/tree/master/charts/access/jira).

This reference details available values for the `teleport-plugin-jira` chart.

(!docs/pages/includes/backup-warning.mdx!)

## `teleport.address`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | Yes |

> This parameter contains the host/port combination of the teleport auth server.

Reviewer: ```suggestion
This parameter contains the host/port combination of the Teleport Auth Service or Proxy Service.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  teleport:
    address: "teleport.example.com:3025"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set teleport.address="teleport.example.com:3025"
  ```
  </TabItem>
</Tabs>

## `teleport.identitySecretName`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | Yes |

> Name of the Kubernetes secret that contains the credentials for the connection.

Reviewer: ```suggestion
Name of the Kubernetes secret that contains the credentials for the connection to the Auth Service.
```

The secret should be in the following format:

```yaml
apiVersion: v1
kind: Secret
type: Opaque
metadata:
  name: teleport-plugin-jira-identity
data:
  auth_id: ...
```

> Check out the [Jira Helm Chart documentation](https://github.com/gravitational/teleport-plugins/tree/master/charts/access/jira#prerequisites) for more information about how to acquire these credentials!

Reviewer: 1. I don't think we need an exclamation here.
2. Should this link to the new docs guide instead? Otherwise we need to maintain two sources of truth.

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  teleport:
    identitySecretName: "teleport-plugin-jira-identity"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set teleport.identitySecretName="teleport-plugin-jira-identity"
  ```
  </TabItem>
</Tabs>

## `teleport.identitySecretPath`

| Type | Default value | Required? |
| - | - | - |
| `string` | `"auth_id"` | No |

> Name of the key in the Kubernetes secret that holds the credentials for the connection. If the secret follows the format above, it can be omitted.

Reviewer: ```suggestion
Name of the key in the Kubernetes secret that holds the credentials for the connection to the Auth Service. If the secret follows the format above, it can be omitted.
```

Reviewer: Also, what does this mean by "the format above"?

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  teleport:
    identitySecretPath: "auth_id"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set teleport.identitySecretPath="auth_id"
  ```
  </TabItem>
</Tabs>

## `jira.url`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | Yes |

> Base URL of the Jira instance

Reviewer: ```suggestion
Base URL of the Jira instance.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  jira:
    url: "https://jira.example.com/"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set jira.url="https://jira.example.com/"
  ```
  </TabItem>
</Tabs>

## `jira.token`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | No |

> Jira token of the bot user to impersonate when sending the messages. Ignored when `jira.apiTokenFromSecret` is set.

Reviewer: ```suggestion
Jira token of the bot user to impersonate when sending Access Request messages. Ignored when `jira.apiTokenFromSecret` is set.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  jira:
    token: "jiraapitoken"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set jira.token="jiraapitoken"
  ```
  </TabItem>
</Tabs>

> ## `jira.apiTokenFromSecret`

Reviewer: I think it would help users navigate this guide more easily if we were to alphabetize the field names

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | No |

Secret containing the Jira token of the bot user.

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  jira:
    apiTokenFromSecret: "teleport-jira-plugin-token"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set jira.apiTokenFromSecret="teleport-jira-plugin-token"
  ```
  </TabItem>
</Tabs>

## `jira.apiTokenSecretPath`

| Type | Default value | Required? |
| - | - | - |
| `string` | `"jiraApiToken"` | No |

Key where the token is located inside the secret specified by `jira.apiTokenFromSecret`.

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  jira:
    apiTokenSecretPath: "token"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set jira.apiTokenSecretPath="token"
  ```
  </TabItem>
</Tabs>

## `jira.username`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | Yes |

> Jira username / email address associated with the token.

Reviewer: ```suggestion
Jira username or email address associated with the token.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  jira:
    username: "user@example.com"
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set jira.username="user@example.com"
  ```
  </TabItem>
</Tabs>

## `jira.recipients`

| Type | Default value | Required? |
| - | - | - |
| `array` | `[]` | Yes |

> List of recipients who will receive notifications about access requests.

Reviewer: ```suggestion
List of recipients who will receive notifications about Access Requests.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  jira:
    recipients:
      - user1@example.com
      - user2@example.com
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set jira.recipients[0]="user1@example.com",jira.recipients[0]="user1@example.com"
  ```
  </TabItem>
</Tabs>

## `jira.project`

| Type | Default value | Required? |
| - | - | - |
| `string` | `""` | Yes |

> List of project who will receive notifications about access requests.

Reviewer: ```suggestion
List of project who will receive notifications about Access Requests.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  jira:
    project: MYPROJ
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set jira.project=MYPROJ
  ```
  </TabItem>
</Tabs>

## `jira.issueType`

| Type | Default value | Required? |
| - | - | - |
| `string` | `"Task"` | Yes |

> List of issueType who will receive notifications about access requests.

Reviewer: Is this the caption you intended to include?

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  jira:
    issueType: Task
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set jira.issueType=Task
  ```
  </TabItem>
</Tabs>

## `log.output`

| Type | Default value | Required? |
| - | - | - |
| `string` | `stdout` | No |

> Logger output. Could be "stdout", "stderr" or a file name, eg. "/var/log/teleport/jira.log"

Reviewer: ```suggestion
Logger output. Can be `stdout`, `stderr` or a file name, eg. `/var/log/teleport/jira.log`.
```

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  log:
    output: /var/log/teleport/jira.log
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set log.output="/var/log/teleport/jira.log"
  ```
  </TabItem>
</Tabs>

## `log.severity`

| Type | Default value | Required? |
| - | - | - |
| `string` | `stdout` | No |

> Logger severity. Possible values are "INFO", "ERROR", "DEBUG" or "WARN".

Reviewer: Let's use code style for the log level names
